url2f44efcfa6_wangyudong_sohutw_java_sdk_2_0_tgz-pJ8-com_sohu_t_api_sdk_json_TestJ8/src/com/sohu/t/api/sdk/data/Counts.java

```diff
@@ -0,0 +1,89 @@
+package com.sohu.t.api.sdk.data;
+
+import com.sohu.t.api.sdk.TBlogException;
+import com.sohu.t.api.sdk.http.Response;
+import com.sohu.t.api.sdk.json.JSONArray;
+import com.sohu.t.api.sdk.json.JSONException;
+import com.sohu.t.api.sdk.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * User: chenxiaojian
+ * Date: 13-1-6
+ * Time: 下午2:52
+ */
+public class Counts {
+    private Long id = null;
+    private int comments_count;
+    private int transmit_count;
+
+
+    public Counts(Response res) throws TBlogException {
+        JSONObject json = res.asJSONObject();
+        init(json);
+    }
+
+    public Counts(JSONObject json) throws TBlogException{
+        init(json);
+    }
+
+    private void init(JSONObject json) throws TBlogException {
+        try {
+            this.comments_count = json.getInt("comments_count");
+            this.transmit_count = json.getInt("transmit_count");
+        } catch (JSONException je) {
+            throw new TBlogException(je.getMessage() + ":" + json.toString(), je);
+        }
+    }
+
+    public static List<Counts> constructStatuses(Response res) throws TBlogException {
+
+        return constructStatuses(res.asJSONArray());
+    }
+
+
+    public static List<Counts> constructStatuses(JSONArray jsonArray) throws TBlogException {
+
+        try {
+            int size = jsonArray.length();
+            List<Counts> countsList = new ArrayList<Counts>(size);
+            for (int i = 0; i < size; i++) {
+                countsList.add(new Counts(jsonArray.getJSONObject(i)));
+            }
+            return countsList;
+        } catch (JSONException jsone) {
+            throw new TBlogException(jsone);
+        } catch (TBlogException te) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)